### PR TITLE
fix(project-url): enforce use of the new url throughout the modules

### PR DESCRIPTION
**Pull request by marlosin**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows [our conventions](../CONTRIBUTING.md#commit-message-format)
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[X] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Modules like UMS, token manager and file uploads through filesets are still using the legacy URL without the project zone.

## What is the new behavior?
All modules will get the base url through a common function.

## Does this PR introduce a breaking change?

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/config/config.ts [hunk 1]
```diff
@@ -31,3 +32,20 @@ export const API = {
 
 /* default refresh token interval: 1 hour and 50 minutes; JEXIA tokens expire in 2 hours */
 export const DELAY = 1000 * 60 * 110;
+
+/**
+ * The default project zone
+ */
+export const DEFAULT_PROJECT_ZONE = "nl00";
```
Comment (KutsenkoA) [inline src/config/config.ts:39]: Do you really need to export default project zone?

Comment (marlosin) [reply src/config/config.ts:39]: yes, for testing purposes 😇

File: src/api/core/tokenManager.ts [hunk 2]
```diff
@@ -259,7 +254,7 @@ export class TokenManager {
    * @ignore
    */
   private get url(): string {
-    return `${API.PROTOCOL}://${this.projectId}.${API.HOST}.${API.DOMAIN}:${API.PORT}`;
+    return getApiUrl(this.config || {});
```
[the remark this answers is not in the document]
Comment (marlosin) [reply src/api/core/tokenManager.ts]: did this mainly because of tests, but I changed them and removed `|| {}`

**Pull request merged**